independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ResteasyReactiveRequestContext.java

```diff
@@ -470,13 +471,17 @@ public ResteasyReactiveRequestContext setRequestUri(URI requestURI) {
         this.path = requestURI.getPath();
         this.authority = requestURI.getRawAuthority();
         this.scheme = requestURI.getScheme();
-        // FIXME: it's possible we may have to also update the query part
+        setQueryParamsFrom(requestURI.toString());
         // invalidate those
         this.uriInfo = null;
         this.absoluteUri = null;
         return this;
     }
 
+    protected void setQueryParamsFrom(String uri) {
+        throw new NotImplementedYet();
```

Comment:
I think this is very dangerous (new implementations of this class won't realize that this method needs to be implemented until they create a new instance of it). I would make this method abstract.

Reply:
Ideally yes, but I don't want to change this important class at this point